templates/components/search/display_data.html.twig

```diff
@@ -23,19 +30,27 @@
          {% endif %}
       </div>
 
+      {% if data['search']['as_map'] == 0 %}
+         {{ include('components/search/table.html.twig') }}
+      {% endif %}
       {% if count > 0 %}
-         {% if data['search']['as_map'] == 0 %}
-            {{ include('components/search/table.html.twig') }}
-         {% endif %}
          {% if data['display_type'] != constant('Search::GLOBAL_SEARCH') and data['search']['as_map'] == 0 %}
             <div class="card-footer search-footer">
                {{ include('components/search/pager.html.twig') }}
             </div>
          {% endif %}
-      {% else %}
+      {% elseif data['search']['as_map'] == 1 %}
```

Comment:
I do not understand this change

Reply:
If the search result view is a table, even if there are no results, the table template is rendered now (table header is hidden if there are no results) so that the JS code doesn't break. It needs the table element and columns intact. If the result view is a map, which is the only other option right now, and there are no results, it displays the "No items found" message. The display of the map when there are results isn't handled anywhere yet and there isn't a template for it.